File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using DevExpress.XtraLayout;
using DevExpress.XtraEditors;


namespace WindowsApplication56 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            // TODO: This line of code loads data into the 'nwindDataSet.Categories' table. You can move, or remove it, as needed.
            this.categoriesTableAdapter.Fill(this.nwindDataSet.Categories);

            dataLayoutControl1.RetrieveFields();

            foreach (BaseLayoutItem baseItem in dataLayoutControl1.Items) {
                LayoutControlItem item = baseItem as LayoutControlItem;
                if (item != null && item.Control.DataBindings.Count > 0)
                    if (item.Control.DataBindings[0].BindingMemberInfo.BindingField == "Description") {
                        dataLayoutControl1.BeginUpdate();
                        Control prevControl = item.Control;
                        Binding binding = prevControl.DataBindings[0];
                        prevControl.DataBindings.Clear();
                        dataLayoutControl1.Controls.Remove(prevControl);
                        Control newControl = new MemoEdit();
                        newControl.Name = "myMemoEdit";
                        // Bind the new control to the same field as the previous control.
                        newControl.DataBindings.Add(new Binding(binding.PropertyName, binding.DataSource,
                            binding.BindingMemberInfo.BindingField, binding.FormattingEnabled));
                        dataLayoutControl1.Controls.Add(newControl);
                        item.Control = newControl;
                        prevControl.Dispose();
                        dataLayoutControl1.EndUpdate();
                        // Change the item's size after the EndUpdate method.
                        item.Size = new Size(100, 80);
                        break;
                    }
            }
        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Imports DevExpress.XtraLayout
Imports DevExpress.XtraEditors


Namespace WindowsApplication56
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			' TODO: This line of code loads data into the 'nwindDataSet.Categories' table. You can move, or remove it, as needed.
			Me.categoriesTableAdapter.Fill(Me.nwindDataSet.Categories)

			dataLayoutControl1.RetrieveFields()

			For Each baseItem As BaseLayoutItem In dataLayoutControl1.Items
				Dim item As LayoutControlItem = TryCast(baseItem, LayoutControlItem)
				If item IsNot Nothing AndAlso item.Control.DataBindings.Count > 0 Then
					If item.Control.DataBindings(0).BindingMemberInfo.BindingField = "Description" Then
						dataLayoutControl1.BeginUpdate()
						Dim prevControl As Control = item.Control
						Dim binding As Binding = prevControl.DataBindings(0)
						prevControl.DataBindings.Clear()
						dataLayoutControl1.Controls.Remove(prevControl)
						Dim newControl As Control = New MemoEdit()
						newControl.Name = "myMemoEdit"
						' Bind the new control to the same field as the previous control.
						newControl.DataBindings.Add(New Binding(binding.PropertyName, binding.DataSource, binding.BindingMemberInfo.BindingField, binding.FormattingEnabled))
						dataLayoutControl1.Controls.Add(newControl)
						item.Control = newControl
						prevControl.Dispose()
						dataLayoutControl1.EndUpdate()
						' Change the item's size after the EndUpdate method.
						item.Size = New Size(100, 80)
						Exit For
					End If
				End If
			Next baseItem
		End Sub
	End Class
End Namespace